dotcom-rendering/src/model/enhanceTableOfContents.ts

```diff
@@ -0,0 +1,60 @@
+import { JSDOM } from 'jsdom';
+import type { TableOfContents, TableOfContentsItem } from 'src/types/frontend';
+
+const isHeading = (element: CAPIElement): boolean => {
+	return (
+		element._type ==
+		'model.dotcomrendering.pageElements.SubheadingBlockElement'
+	);
+};
+
+const isH2 = (element: CAPIElement): element is SubheadingBlockElement => {
+	console.log('isheading', element, isHeading(element));
+	return isHeading(element);
+};
+
+const extractText = (element: SubheadingBlockElement): string => {
+	const frag = JSDOM.fragment(element.html);
+	if (!frag.firstElementChild) return '';
+	return frag.textContent?.trim() ?? '';
+};
+
+const hasInteractiveContentsElement = (blocks: Block[]): boolean => {
+	return !!blocks.find((block) =>
+		block.elements.find(
+			(element) =>
+				element._type ===
+				'model.dotcomrendering.pageElements.InteractiveContentsBlockElement',
+		),
+	);
```

Comment:
Might be more semantic to use `some` here?

```ts
	return blocks.some((block) =>
		block.elements.some(
			(element) =>
				element._type ===
				'model.dotcomrendering.pageElements.InteractiveContentsBlockElement',
		),
	);
```

Reply:
yep!